examples/models/question_answering/QuestionAnswering.py

```diff
@@ -0,0 +1,495 @@
+        self.encoding_model = None
+        self.torch_device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        self.qa_tokenizer = tokenizer 
+
+    def _create_model(self, scratch: bool):
+        qa_model = AutoModelForQuestionAnswering.from_pretrained(BERT_SQUAD);
+        qa_model = qa_model.to(self.torch_device)
+        qa_model.eval()
+        return qa_model
+
+    def train(self, dataset_path=None, shared_params=None, **train_args):
+        pass
+
+    def evaluate(self, dataset_path):
+        return float(1)
+
+
+    def dump_parameters(self):
+        params = {}
+        return params
+
+    def load_parameters(self, params):
+        fine_tune_dataset_path = params
+        print (fine_tune_dataset_path)
+        print (type(fine_tune_dataset_path),'*************')
+        with open(fine_tune_dataset_path, 'rb') as f:
+            zip_file_base64 = f.read()
+
+        with tempfile.NamedTemporaryFile() as tmp:
+            # Convert back to bytes & write to temp file
+            with open(tmp.name, 'wb') as f:
+                f.write(zip_file_base64)
+            with tempfile.TemporaryDirectory() as root_path:
+                dataset_zipfile = zipfile.ZipFile(tmp.name, 'r')
+                dataset_zipfile.extractall(path=root_path)
+                dataset_zipfile.close()
+
+                metadata_path = "{}/metadata.csv".format(root_path)
+                all_json = glob.glob("{}/arxiv/**/*.json".format(root_path), recursive=True)
+
+                self.df_topic_only = DataLoader.json_load_to_df(
+                    all_json=all_json,
+                    metadata_path=metadata_path, topic_words=self.topic_words, topic_year=self.topic_year)
+
+                self.topic_encoded, self.encoding_model = DataLoader.load_to_embedding(self.df_topic_only)
+
+        self._model = self._create_model(scratch=False)
+
+    def predict(self, queries: List[Any]) -> List[Any]:
+        predictions = list()
+        data = queries[0]
+        assert isinstance(data, dict)
+
+        try:
+            task_questions = data['questions']
+        except:
+            task_questions = []
+            for question_set in data.values():
+                task_questions.extend(question_set['questions'])
+
+        dict_queries = []
+        for idx, query in enumerate(task_questions):
+            print(f"Getting answers from query: {query}")
+            answers_list = []
+            if '\n' in query:
+                '''this is for queries with /n separated context'''
+                question, query_paragraph = query.splitlines()
+                answer, conf, paragraph = self.__answerQuestion(question, query_paragraph, self._model)  # best_answer, confidence, best_paragraph
+                if answer:
+                    start, end = self.__findStartEndIndexSubstring(paragraph, answer)
+                    answer_parag = paragraph[start:end]
+                else:
+                    answer_parag = 'idk'
```

Comment:
what is idk?

Reply:
idk stands for "I don't know", in certain situation, the model cannot find answers for queries.

Reply:
Pls make this kind of message clear. What does "I don't know" mean?  why the model cannot find answers?

Reply:
when the model does not have related knowledge to the query question, model won't make prediction thus it will give 'I don't know' as an answer.